packages/sitecore-jss/src/native-fetcher.test.ts

```diff
@@ -7,10 +7,18 @@ import debug from './debug';
 
 use(spies);
 
-let fetchInput: URL | RequestInfo | undefined;
+let fetchInput: RequestInfo | URL | undefined;
 let fetchInit: RequestInit | undefined;
 
-const mockFetch = (status: number, response: unknown = {}, jsonError?: string) => {
+const mockFetch = (
+  status: number,
+  response: unknown = {},
+  {
+    jsonError,
+    textError,
+    responseType,
+  }: { jsonError?: string; textError?: string; responseType?: 'text' | 'json' } = {}
+) => {
```

Comment:
can you please add a test case to ensure native fetcher returns correct headers in its response?